### DISCUSSION TITLE
Maui & MIDI

### DISCUSSION THREAD

**Discussion by buzzripper** [dotnet/maui#4695] [General]
I'd like to create a MAUI application that deals with midi - connecting to ports, sending/receiving midi messages, etc. Is there a library that would work with MAUI, or any plans to add one? I can't find any reference to MIDI anywhere.

Thanks

**Comment by francoistanguay** [comment]
Unrelated but @MartinZikmund  did a great job supporting the MIDI api in Uno (short video here: https://blog.mzikmund.com/2020/08/unoconf-2020/ ), so maybe you can look at his x-platform implementation of https://platform.uno/docs/articles/features/windows-devices-midi.html  for inspiration.

**Reply by buzzripper** [reply]
I don't think I'm up to writing my own, but thank you for the links.

**Reply by drdream** [reply]
I was all ready to complain.. But @buzzripper .. @francoistanguay handed us the keys.. They provide an API that automagically works to detect audio devices, and even detected my Midi Bus on OSX.. Async/Await for midi message sending.. What more do you need?

**Reply by MartinZikmund** [reply]
![BugsBunnyGIF](https://user-images.githubusercontent.com/1075116/182619108-ce2389ad-da6e-4817-a749-a58b1754dc24.gif)

**Reply by leosimanonok** [reply]
@drdream Could you link to the API? I still cant seem to find anything that works of mac for .net maui
